cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -395,15 +458,26 @@ func doInstall(mst *initramfsMountsState, model *asserts.Model, sysSnaps map[sna
 		return err
 	}
 
-	currentSeed, err := mst.LoadSeed(mst.recoverySystem)
+	if preseed {
+		runMode := false
+		if err := installApplyPreseededData(preseedSeed,
+			boot.InitramfsWritableDir(model, runMode)); err != nil {
+			return err
+		}
+	}
+
+	// Create drivers tree mount units
```

Comment:
This comment is not very helpful either - i think a comment about the expected behaviour in this case and how it's consumed would be much more helpful

Reply:
tweaked a bit